ROADMAP.md

```diff
@@ -0,0 +1,41 @@
+# `sget` Roadmap
+
+### Today
+
+`sget` fetches blobs from OCI registries, using Docker API registry auth, checking content digests to ensure integrity, and verifying any `cosign` signatures that are found in the registry.
+
+`sget` currently has no functionality to fetch arbitrary URLs.
+
+### Phase 0
+
+Add basic URL-fetching functionality to `sget`, with an optional ability to check content digests to ensure integrity.
+
+This phase may also include a manual verification step where buffered contents are shown to the user, so they can manually inspect the contents before proceeding.
+
+### Phase 1: "Social Proof"
+
+In the absence of a maintainer-provided policy, `sget` users can gain assurance about the safety of an artifact by relying on crowd-sourced verificatioin and signatures.
+
+Signatures will be stored in Rekor, using certificates created by Fulcio's "keyless" OIDC capabilities.
+
+Before trusting an artifact, users should be able to:
+
+- see the total number of unique identities that have signed indicating that the artifact is safe to fetch.
+- see a random small subset of those identities.
+- see any identities with email addresses that match the URL's domain.
+- locally configure trusted identities, and trust those identities more than arbitrary random identities.
+- locally configure policies, and automatically trust artifacts that satisfy those requirements.
```

Comment:
What's an example of such a policy?

Depending on how complex we see this being, I'm wondering if it makes sense to push this to a later phase (or a "Phase 1.5")...

Reply:
Yeah good question. I was imagining some kind of local policy like "I trust `A`, `B` and `C`, but always want to require 2+ of their signatures before proceeding"

I can move this to a 1.5 subsection if it helps. It's probably closely related to maintainer policies, at least in terms of the policy language definition.

Reply:
Agreed on "step 1.5", this is important to get right and will be a fair bit of work and doesn't block any of the rest.

I think TBH there's a need for this in the sigstore ecosystem more broadly; *every* downstream (cosign, sget, package repositories) should be able to express policies. (Not that we need to solve it once and for all in sget).

Reply:
1.5 it is!

Re: sigstore-wide policy unification, I agree, I think there's been a bit of that being thrown around in `cosigned` webhook task force meetings, maybe we can ~~steal~~ borrow what they've come up with.